test/KillSwitchOracle.t.sol

```diff
@@ -0,0 +1,393 @@
+            liquidationBonus:     110_00
+        });
+
+        // Paused asset
+        _initReserve({
+            asset:                asset4,
+            active:               true,
+            frozen:               false,
+            paused:               true,
+            ltv:                  80_00,
+            liquidationThreshold: 83_00,
+            liquidationBonus:     105_00
+        });
+
+        // Inactive asset
+        _initReserve({
+            asset:                asset5,
+            active:               false,
+            frozen:               false,
+            paused:               false,
+            ltv:                  0,
+            liquidationThreshold: 0,
+            liquidationBonus:     0
+        });
+
+        assertEq(pool.getReservesList().length, 5);
+
+        vm.prank(owner);
+        killSwitchOracle.setOracle(address(oracle), 0.99e8);
+        oracle.setLatestAnswer(0.98e8);
+
+        vm.expectEmit(address(killSwitchOracle));
+        emit Trigger(address(oracle), 0.99e8, 0.98e8);
+        vm.expectEmit(address(killSwitchOracle));
+        emit AssetLTV0(asset1);
+        vm.expectEmit(address(killSwitchOracle));
+        emit AssetFrozen(asset2);
+        vm.prank(randomAddress);  // Permissionless call
+        killSwitchOracle.trigger(address(oracle));
+
+        _assertReserve({
+            asset:                asset1,
+            active:               true,
+            frozen:               false,
+            paused:               false,
+            ltv:                  0,
+            liquidationThreshold: 83_00,
+            liquidationBonus:     105_00
+        });
+
+        _assertReserve({
+            asset:                asset2,
+            active:               true,
+            frozen:               true,
+            paused:               false,
+            ltv:                  0,
+            liquidationThreshold: 0,
+            liquidationBonus:     0
+        });
+
+        _assertReserve({
+            asset:                asset3,
+            active:               true,
+            frozen:               true,
+            paused:               false,
+            ltv:                  0,
+            liquidationThreshold: 25_00,
+            liquidationBonus:     110_00
+        });
+
+        _assertReserve({
+            asset:                asset4,
+            active:               true,
+            frozen:               false,
+            paused:               true,
+            ltv:                  80_00,
+            liquidationThreshold: 83_00,
+            liquidationBonus:     105_00
+        });
+
+        _assertReserve({
+            asset:                asset5,
+            active:               false,
+            frozen:               false,
+            paused:               false,
+            ltv:                  0,
+            liquidationThreshold: 0,
+            liquidationBonus:     0
+        });
```

Comment:
I'd use this patter where we create a struct with all the values for the "before" assertions and use the same structs for "after" assertions, swapping values that should change. It'd be much more readable.
Even if it would be sth like:
```
        _assertReserve({
            asset:                initialConfig.asset,
            active:               initialConfig.active,
            frozen:               initialConfig.frozen,
            paused:               initialConfig.paused,
            ltv:                  0,
            liquidationThreshold: initialConfig.liquidationThreshold,
            liquidationBonus:     initialConfig.liquidationBonus
        });
```

Reply:
Good call. I refactored it.